cashctrl_ledger/tests/test_accounts.py

```diff
@@ -14,165 +14,139 @@
-            ledger.modify_account(
-                account=1147,
-                currency="CHF",
-                description="test account",
-                tax_code="MwSt. 2.6%",
-                group="/Assets/Anlagevermögen",
-            )
-
-    def test_modify_account_with_invalid_currency_error(self, ledger):
+            engine.accounts.modify({
+                "account": 1147,
+                "currency": "CHF",
+                "description": "test account",
+                "tax_code": "MwSt. 2.6%",
+                "group": "/Assets/Anlagevermögen",
+            })
+
+    def test_modify_account_with_invalid_currency_error(self, engine):
         with pytest.raises(ValueError):
-            ledger.modify_account(
-                account=1148,
-                currency="not-existing-currency",
-                description="test account",
-                tax_code=None,
-                group="/Assets/Anlagevermögen",
-            )
-
-    def test_modify_account_with_invalid_tax_raise_error(self, ledger):
+            engine.accounts.modify({
+                "account": 1148,
+                "currency": "not-existing-currency",
+                "description": "test account",
+                "tax_code": None,
+                "group": "/Assets/Anlagevermögen",
+            })
+
+    def test_modify_account_with_invalid_tax_raise_error(self, engine):
         with pytest.raises(ValueError):
-            ledger.modify_account(
-                account=1149,
-                currency="USD",
-                description="test create account",
-                tax_code="Non-Existing Tax Code",
-                group="/Assets/Anlagevermögen",
-            )
-
-    def test_modify_account_with_invalid_group_raise_error(self, ledger):
+            engine.accounts.modify({
+                "account": 1149,
+                "currency": "USD",
+                "description": "test create account",
+                "tax_code": "Non-Existing Tax Code",
+                "group": "/Assets/Anlagevermögen",
+            })
+
+    def test_modify_account_with_invalid_group_raise_error(self, engine):
         with pytest.raises(ValueError):
-            ledger.modify_account(
-                account=1149,
-                currency="USD",
-                description="test create account",
-                tax_code="MwSt. 2.6%",
-                group="/ABC",
-            )
-
-    def test_mirror_accounts_with_root_category(self, ledger):
+            engine.accounts.modify({
+                "account": 1149,
+                "currency": "USD",
+                "description": "test create account",
+                "tax_code": "MwSt. 2.6%",
+                "group": "/ABC",
+            })
+
+    def test_mirror_accounts_with_root_category(self, engine):
         """This test ensures that root categories remain untouched, and all new accounts categories
         expected to be created are done so before any existing accounts are mirrored.
```

Comment:
I haven't really understood the motivation for the test (why do we need it, what would me miss if it was not there)?